Formula/aws-sdk-cpp.rb

```diff
@@ -0,0 +1,50 @@
+require "set"
+
+class AwsSdkCpp < Formula
+  desc "AWS SDK for C++"
+  homepage "https://github.com/aws/aws-sdk-cpp"
+  url "https://github.com/aws/aws-sdk-cpp/archive/0.10.6.tar.gz"
+  sha256 "01ad50677e2bde3bcd51128a9e043f76371e9642a5bea235637ce8e450eec5de"
+  head "https://github.com/aws/aws-sdk-cpp.git"
+
+  option "with-custom-memory-management", "Build with custom memory management"
+
+  option "with-static", "Build with static linking"
+
+  option "without-http-client", "Don't include the libcurl HTTP client"
+
+  depends_on "cmake" => :build
+
+  def install
+    custom_memory_management = (build.with? "custom-memory-management") ? 1 : 0
+    static_linking = (build.with? "static") ? 1 : 0
+    without_http_client = (build.without? "http-client") ? 1 : 0
```

Comment:
I don't think we need to split these out up here since it's one to one.
args << "-DCUSTOM_MEMORY_MANAGEMENT=1" if build.with? "custom-memory-management"
etc.